include/uart.h

```diff
@@ -99,6 +99,11 @@ enum uart_event_type {
 	 * - When RX timeout occurred, and data was stored in provided buffer.
 	 *   This can happen multiple times in the same buffer.
 	 * - When provided buffer is full.
+	 *
+	 * The data represented by the event is stored in
```

Comment:
i would move this clarification to event structure. Here user is not aware of struct fields so it goes to much into details when it's a place to explain what given event is.

Also this phrase ` instead of being an absolute size.` is in my opinion redundant and confusing since it's hard to tell what `absolute size` means here.

Reply:
Done.